pagefaultgames/pokerogue
# [Bug] Fixes Freezy Frost visual bug when attacking ally, and Freezy Frost not applying effects when fainting a target

## What are the changes the user will see?

Freezy Frost properly updates stat info when attacking allies.
Freezy Frost properly applies effects to all pokemon on the field, even when the target is fainted using the move.

## Why am I making these changes?



bugs

## What are the changes from a developer perspective?


The `resetStats()` helper function now returns a `Promise<void>` from `updateInfo()`. The array of promises is then resolved in the apply function. This resolves the visual bug when attacking an ally with the move (left mon attacks right mon seems to be the only issue). Also utilizes the async/await pattern instead of purely Promises.
Removed the condition for `super.apply()` which prevents the move from applying effects to the entire field when the target faints. 

### Screenshots/Videos


[See Discord report for bug showcases](https://github.com/pagefaultgames/pokerogue/pull/3753#issuecomment-2308927883)

Attacking ally properly updates visual info

https://github.com/user-attachments/assets/7f09a2db-467b-4947-9ce0-54a763d0902f

Fainting enemy properly reduces stats of all mons on field

https://github.com/user-attachments/assets/7d82a866-23a0-46ce-8cf2-038b23376c72

## How to test the changes?




Set battle type to double, give all mons a stat raising move, give your mons freezy frost/haze/clear smog (all affected moves), see that info properly updates/see that effects are applied despite fainting a target.

## Checklist
- [n/a] If I have text, did I add make it translatable and added a key in the English language?
